services/core-api/app/api/mines/notice_of_departure/models/notice_of_departure.py

```diff
@@ -69,7 +70,10 @@ def create(cls,
             nod_title=nod_title,
             nod_description=nod_description,
             nod_type=nod_type,
-            nod_status=nod_status)
+            nod_status=nod_status,
+            nod_no=text(
+                f'\'NOD-\' || (SELECT permit_no FROM permit WHERE permit_guid = \'{permit.permit_guid}\') || \'-\' || (SELECT TO_CHAR((SELECT COUNT(*) FROM notice_of_departure nod WHERE permit_guid = \'{permit.permit_guid}\') + 1, \'fm00\'))'
```

Comment:
Nice!

Reply:
Sorry, just checking this. It seems a SQL injection can be used and become dangerous. We rely always on ORM, the use of SQL Alchemy would be very advisable to avoid this security vulnerability.

Reply:
This is not directly getting any values from the frontend. Instead, we are generating a value from within the database. the passed value is coming from the application itself. My thought the risk of sql injection is minimal but I am open to seeing how we can use sqlalchemy to improve this.

Reply:
The real risk of SQL injection is that the hacker can replace your SELECT for something else such as DELETE or UPDATE. This is the real danger here.

Reply:
After chatting with @areyeslo we realized that the chance of SQL injection is low as we are dealing with generating queries using validated data. However, we realize it can be uncomfortable to have raw queries. I will do a new PR to use SQL alchemy helper functions.